scripts/masternode.js

```diff
@@ -2,13 +2,19 @@
 
 class Masternode {
     static protocolVersion = 70926;
-    constructor({walletPrivateKey, mnPrivateKey, collateralTxId, outidx, addr} = {}) {
-	this.walletPrivateKey = walletPrivateKey;
+    constructor({walletPrivateKeyPath, mnPrivateKey, collateralTxId, outidx, addr} = {}) {
+	//this.walletPrivateKey = walletPrivateKey;
+	this.walletPrivateKeyPath = walletPrivateKeyPath;
 	this.mnPrivateKey = mnPrivateKey;
 	this.collateralTxId = collateralTxId;
 	this.outidx = outidx;
 	this.addr = addr;
     }
+	async getWalletPrivateKey(){
+		let result= ((!masterKey.isHD) ? (await masterKey.getPrivateKey()) : (await masterKey.getPrivateKey(this.walletPrivateKeyPath)))
```

Comment:
The ternary operation is not needed, as non-hd keys ignore the `path` parameter; however I think it could stay for readability's sake

Reply:
I think it's actually easier to follow / read without the ternary at all (since the wallet class logic handles this fine, this is essentially "wasted cycles" and one can assess quickly via the Masterkey classes that this logic is safe).

Reply:
Agree I will change soon